v0/destinations/lambda/transform.js

```diff
@@ -0,0 +1,125 @@
+const _ = require("lodash");
+const {
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  CustomError
+} = require("../../util");
+
+const DEFAULT_INVOCATION_TYPE = "Event"; // asynchronous invocation
+const MAX_PAYLOAD_SIZE_IN_KB = 256; // only for asynchronous invocation
+
+// Returns a transformed payload, after necessary property/field mappings.
+function process(event) {
+  if (!event.destination.Config) {
+    throw new CustomError("destination.Config cannot be undefined", 400);
+  }
+  return {
+    payload: JSON.stringify(event.message),
+    destConfig: event.destination.Config
+  };
+}
+
+// Returns the size of the data in kilobytes
+function getSizeInKB(data) {
+  const size = Buffer.byteLength(JSON.stringify(data));
+  return size / 1024;
+}
+
+// Returns an array of payloads within the size limit
+function getBatchedPayloads(inputs, maxBatchSize) {
+  const batchedPayloads = [];
+  let payloadChunk = [];
+  let chunkMetadata = [];
+  const errorMetadata = [];
+  inputs.forEach(input => {
+    if (getSizeInKB([input.message]) > MAX_PAYLOAD_SIZE_IN_KB) {
+      errorMetadata.push(input.metadata);
+      return;
+    }
+    if (
+      payloadChunk.length.toString() === maxBatchSize ||
```

Comment:
maxBatchSize could be int right?

Reply:
this logic seems fine but looks quite intensive compute, if we compute size of each input and then add them up till MAX_PAYLOAD_SIZE_IN_KB wiil be efficient but we need to consider overhead for array wrapping if we set MAX_PAYLOAD_SIZE_IN_KB to 250 then those 6 KB will be cover those extra things.

Reply:
Updated

Reply:
Rather than setting `MAX_PAYLOAD_SIZE_IN_KB ` to 250KB, I stored the size of `[input.message]` instead of `input.message`. This way, the sum of the sizes will just be greater than the size of array of inputMessages.